services.py

```diff
@@ -70,3 +72,28 @@ def update_or_create_criterion(data, criterion=None):
         criterion.deactivate()
 
     return criterion.save()
+
+
+def state_information(state):
+    links = Link.query.filter_by(state=state, active=True).all()
+
+    # https://stackoverflow.com/questions/58333162/how-to-get-top-n-results-per-group-from-a-pool-of-ids-in-sqlalchemy
+    subquery = db.session.query(
+        Score,
+        func.rank().over(
+            order_by=Score.created_at.desc(),
+            partition_by=Score.criterion_id
+        ).label('rank')
+    ).filter(Score.state == state).subquery()
```

Comment:
This is the best way I figured out to return the most recent score for each criterion that the state has.

Reply:
This is neat!